### PR TITLE
Specialist Bug Fixes + Rocket Launcher Reloads

### PR DESCRIPTION
**Pull request by Surrealaser**
:cl: Surrealistik
fix: Bug with WP rockets fixed; they will now create areas of fire as intended.
fix: B18 grenade belt now holds an appropriate amount of grenades.
tweak: The rocket launcher can now be unloaded; this takes half as much time as loading (3 seconds).
tweak: The rocket launcher doesn't start with a rocket loaded; as compensation the demolitionist gets an additional WP rocket in its starting kit.
tweak: Plastique in demo kit replaced with detpacks.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 1]
```diff
@@ -708,35 +716,43 @@
...
+			else
 				playsound(user,'sound/machines/click.ogg', 25, 1)
 		else
 			to_chat(user, "<span class='warning'>Your reload was interrupted!</span>")
 			return
 	else
-		rocket.loc = get_turf(src)
-		replace_ammo(,rocket)
+		loaded_rocket.loc = src
+		replace_ammo(,loaded_rocket)
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]:
```suggestion
		replace_ammo(null,loaded_rocket)
```

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 2]
```diff
@@ -708,35 +716,43 @@
...
 		current_mag.current_rounds = current_mag.max_rounds
-		rocket.current_rounds = 0
-	rocket.update_icon()
 	return TRUE
 
 /obj/item/weapon/gun/launcher/rocket/unload(mob/user)
 	if(user)
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: change this to early return

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 3]
```diff
@@ -708,35 +716,43 @@
...
-		if(!current_mag.current_rounds) 
+		if(!current_mag.current_rounds)
 			to_chat(user, "<span class='warning'>[src] is already empty!</span>")
-		else 							
-			to_chat(user, "<span class='warning'>It would be too much trouble to unload [src] now. Should have thought ahead!</span>")
+		else
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: throw a return in the above if and remove this else

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 4]
```diff
@@ -701,35 +708,44 @@
...
+	if(!user)
+		return
+	if(!current_mag.current_rounds)
+		to_chat(user, "<span class='warning'>[src] is already empty!</span>")
+		return
+	to_chat(user, "<span class='notice'>You begin unloading [src].</span>")
+	if(!do_after(user,current_mag.reload_delay * 0.5, TRUE, 5, BUSY_ICON_FRIENDLY))
+		to_chat(user, "<span class='warning'>Your unloading was interrupted!</span>")
+		return
+	current_mag.current_rounds = 0
```
Comment (DominikPanic) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: What if the rocket launcher has more than one round? For example the `m57a4`

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 5]
```diff
@@ -620,7 +620,9 @@
 
 	flags_gun_features = GUN_INTERNAL_MAG|GUN_WIELDED_FIRING_ONLY
 	gun_skill_category = GUN_SKILL_SPEC
+	reload_sound = 'sound/weapons/gun_mortar_reload.ogg'
 	var/datum/effect_system/smoke_spread/smoke
+	var/obj/item/ammo_magazine/rocket/loaded_rocket = null
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: What's the difference between this and `current_mag` aka `/obj/item/ammo_magazine/internal/launcher/rocket`?

Comment (Surrealaser) [reply code/modules/projectiles/updated_projectiles/guns/specialist.dm]: The internal magazine is essentially a magazine of magazines.

Comment (Rohesie) [reply code/modules/projectiles/updated_projectiles/guns/specialist.dm]: Why making another magazine var then?

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 6]
```diff
@@ -799,52 +809,26 @@
...
-
-/obj/item/weapon/gun/launcher/rocket/reload(mob/user, obj/item/ammo_magazine/rocket)
-	if(flags_gun_features & GUN_BURST_FIRING)
+/obj/item/weapon/gun/launcher/rocket/unload(mob/user)
+	if(!user)
 		return
-
-	if(!rocket || !istype(rocket) || rocket.caliber != current_mag.caliber)
-		to_chat(user, "<span class='warning'>That's not going to fit!</span>")
+	if(!current_mag || isnull(current_mag) || current_mag.loc != src)
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]:
```suggestion
	if(!current_mag || current_mag.loc != src)
```
`isnull(A)` will always be false if `!A` is false.

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm [hunk 7]
```diff
@@ -799,52 +809,26 @@
 		current_mag.current_rounds++
 	return TRUE
 
-
-/obj/item/weapon/gun/launcher/rocket/reload(mob/user, obj/item/ammo_magazine/rocket)
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/guns/specialist.dm]: Can you reload the rocket gun? Have you tested this?

Comment (Surrealaser) [reply code/modules/projectiles/updated_projectiles/guns/specialist.dm]: Yes

File: code/modules/projectiles/updated_projectiles/guns/specialist.dm (hunk 6 above)
Comment (Surrealaser) [reply code/modules/projectiles/updated_projectiles/guns/specialist.dm]: This is how it's set up in the reload proc; I assume it's this way for a reason

**Pull request merged**